### PR TITLE
Take care of FailedPutCount in putRecordBatch resp

### LINKED ISSUE #37: Small amount of event not streamed without errors
Hello

In my pipeline : The data came with kinesis stream, I don't use Dynamodb mapping, only the default, firehose write data into an s3 file.

Yesterday two 'event' were not recorder into s3. I dig into Kinesis Stream and found my two event : 

```ruby
#<struct Aws::Kinesis::Types::Record
  sequence_number="49*3138",
  approximate_arrival_timestamp=2017-02-13 17:18:39 +0100,
  data="3444946|2017-02-13 16:18:39.323|***",
  partition_key="partitionkey">,
#<struct Aws::Kinesis::Types::Record
  sequence_number="49*28130",
  approximate_arrival_timestamp=2017-02-13 17:18:39 +0100,
  data="3444947|2017-02-13 16:18:39.364|***",
  partition_key="partitionkey">,
```
In s3 I have an s3 file with lot's of events. I have one before, and after but not this two : 
```
3444945|2017-02-13 16:18:30.100|***
3444948|2017-02-13 16:18:41.832|**
```

I dig into cloudwatch to find errors in the lambda but without seeing any issues.

Where to look at to understand why records where not inserted into s3? 😞

### PR DESCRIPTION
**Pull request by DaichiUeura**
This is really needed not to lose records when putRecordBatch
doesn't accept all records.

Fix #37

### CODE REVIEW

**Comment by benoittgt** [comment]
Thanks a lot @DaichiUeura

**Comment by benoittgt** [comment]
Tested here in "demo" without issue. I left few comments.

File: constants.js [hunk 1]
```diff
@@ -6,4 +6,6 @@ KINESIS_SERVICE_NAME = "aws:kinesis";
 FIREHOSE_MAX_BATCH_COUNT = 500;
 // firehose max PutRecordBatch size 4MB
 FIREHOSE_MAX_BATCH_BYTES = 4 * 1024 * 1024;
-targetEncoding = "utf8";
\ No newline at end of file
+MAX_RETRY_ON_FAILED_PUT = 3;
```
Comment (benoittgt) [inline constants.js]: What about let the user be able to override them with env variables? 

```javascript
MAX_RETRY_ON_FAILED_PUT = process.env.MAX_RETRY_ON_FAILED_PUT || 3;
RETRY_INTERVAL_MS = process.env.RETRY_INTERVAL_MS || 300;
```

Comment (DaichiUeura) [inline constants.js]: Good suggestion:) I will update the commit later.

**Comment by benoittgt** [comment]
Works perfectly. Had 2 FailedPutCount last week and the retry mechanism perfectly manage them.

I think we should merge it @IanMeyers.

**Comment by IanMeyers** [comment]
Great stuff - thanks all.

**Pull request merged**